Source/WebCore/style/StyleBuilderConverter.h

```diff
@@ -2174,7 +2174,14 @@ inline OptionSet<MarginTrimType> BuilderConverter::convertMarginTrim(BuilderStat
     if (!list || !list->size())
         return RenderStyle::initialMarginTrim();
     OptionSet<MarginTrimType> marginTrim;
-    ASSERT(list->size() <= 4);
+    for (auto& item : *list) {
+        if (item.valueID() == CSSValueBlock)
+            marginTrim.add({ MarginTrimType::BlockStart, MarginTrimType::BlockEnd });
+        if (item.valueID() == CSSValueInline)
+            marginTrim.add({ MarginTrimType::InlineStart, MarginTrimType::InlineEnd });
```

Comment:
Can we just add these to the loop a few lines below?

Reply:
You can't mix block/inline with the other keywords, so I decided to handle them separately in the converter